### ISSUE TITLE
HashMap.getOrElseGet is calling Supplier even if the key is present in the map.

### ISSUE THREAD

**Issue by jijisv** [aol/cyclops#992]
The getOrElseGet method is calling the supplier even if the the key is present in map.
Eg code

```java
        HashMap<String, String> dataMap = HashMap.empty();
        dataMap = dataMap.put("1", "one");
        dataMap = dataMap.put("2", "two");

        System.out.println("Map: " + dataMap);
        String val = dataMap.getOrElseGet("1", () -> {
            System.out.println("Using supplier as key not found...");
            return "ten";
        });

        System.out.println("Value: " + val);
```
The output is
```java
Map: [{1=one}, {2=two}]
Using supplier as key not found...
Value: one
```

As you can see  in the logs above, the Supplier has been invoked even though the key "1" is present in the map.

**Comment by jijisv** [comment]
Fix: https://github.com/aol/cyclops/pull/993/files?w=1

**Comment by johnmcclean** [comment]
Great thanks for the bug report and the fix @jijisv !

**Comment by johnmcclean** [comment]
Released in 10.1.1
